### PR TITLE
add mediaType method

### LINKED ISSUE #19: Media type based on preferences
For my [error transformation](https://github.com/hapijs/hapi/blob/master/API.md#error-transformation) plugin, [hapi-error-page](https://github.com/sholladay/hapi-error-page), I need to be sure to respect the client's desired media type, taking into account its [q weightings](https://tools.ietf.org/html/rfc7231#section-5.3.2).

I expected there to be a method for this, since such a thing exists for all of the other header types, but that doesn't seem to be the case.

Similar to the other methods, the signature would be `mediaType(mediaTypeHeader, [preferences])`.

### PR DESCRIPTION
**Pull request by venikman**
First I did (copied) tests from other headers and made them pass. I read spec. But now sure If I am did right decision about behavior (this is my first time working with any spec).  

If implementation will be approved I will start to write docs.

Close #19

### CODE REVIEW

**Comment by hueniverse** [comment]
@sholladay can you do a review to see if this does what you need?

File: test/mediatype.js [hunk 1]
```diff
@@ -20,6 +20,78 @@ const expect = Code.expect;
...
+
+    it('requires the preferences parameter to be an array', () => {
+
+        expect(() => {
+
+            Accept.mediaType('application/json;q=0.6, text/plain;q=0.8', 'text/plain');
+        }).to.throw('Preferences must be an array');
+    });
+
+    it('returns empty string with header is empty', () => {
```
[the remark this answers is not in the document]
Comment (venikman) [reply test/mediatype.js]: Fixed

File: test/mediatype.js [hunk 2]
```diff
@@ -20,6 +20,78 @@ const expect = Code.expect;
...
+
+        const mediaType = Accept.mediaType('');
+        expect(mediaType).to.equal('*/*');
+    });
+
+    it('returns empty string if header is missing', () => {
```
[the remark this answers is not in the document]
Comment (venikman) [reply test/mediatype.js]: Fixed

File: test/mediatype.js [hunk 3]
```diff
@@ -20,6 +20,78 @@ const expect = Code.expect;
...
+    it('returns first found preference that header includes', () => {
+
+        const mediaType = Accept.mediaType('text/html, application/json, text/plain', ['text/csv', 'application/json']);
+        expect(mediaType).to.equal('application/json');
+    });
+
+    it('returns preference with highest specificity', () => {
+
+        const mediaType = Accept.mediaType('text/*, text/html, application/json', ['text/plain', 'text/html']);
+        expect(mediaType).to.equal('text/plain');
```
[the remark this answers is not in the document]
Comment (venikman) [reply test/mediatype.js]: I look on this that `text/*` and `text/html` just unite into `text/*`

**Pull request merged**